### DISCUSSION TITLE
Windows Workspace Widget

### DISCUSSION THREAD

**Discussion by CasanovaSan** [amnweb/yasb#69] [Ideas]
So i am a big user of the Windows Workspace feature, and really wish that Yasb would get a widget for it that kinda looks like the Komorbi one
![image](https://github.com/user-attachments/assets/66a8b6c1-28ec-46c5-8550-5b2cd9126517)

(basically a widget that looks like this and can change it by clicking the "circle")
![image](https://github.com/user-attachments/assets/0cacd2d4-acce-4a22-8aff-2a11d27d4046)

**Comment by amnweb** [comment]
Unfortunately, there is no public API available to enumerate these virtual desktops, There are some unofficial modules which we can use, but it will be missing animations and MS can break it anytime with any feature update. The public [IVirtualDesktopManager](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-ivirtualdesktopmanager) interface is very limited, it allows you to only query whether a given window belongs to the currently active virtual desktop, and to move a window to a specific virtual desktop. Nothing more.

**Comment by amnweb** [comment]
Done https://github.com/amnweb/yasb/commit/eb2cf38a202818f6c161503b663fa11cb0deadd9

![image](https://github.com/user-attachments/assets/7e045053-9d91-4479-ab65-2b669679bb21)

**Reply by Salua99cs** [reply]
Hi, how do I get this to work?

When I first installed YASB Reborn I was testing a few themes, some of them had these buttons and they worked. 
And now suddenly the don't. They won't show up unless I have komorebi installed, and then it'll only apply to Workspaces rather than my actual desktops. But I prefer Windows' multiple desktops feature so I got rid of komorebi.

I reinstalled YASB Reborn, but the desktop-switching buttons won't show on any of themes I try.
Am I missing something?

**Reply by amnweb** [reply]
Yes, you are missing the documentation to understand how YASB works.

**Reply by Salua99cs** [reply]
Ah, okay, apologies. I guess I just don't understand it.

But... I did end up finding one theme that had Windows Workspaces, and I copied that part over to the config of the other theme I'm using. Though maybe a sloppy work-around... it worked.
